chpl/chpl-api/src/main/java/gov/healthit/chpl/ApiExceptionControllerAdvice.java

```diff
@@ -169,25 +170,52 @@
     }
 
     @ExceptionHandler(CertificationBodyAccessException.class)
-    public ResponseEntity<ErrorResponse> exception(final CertificationBodyAccessException e) {
+    public ResponseEntity<ErrorResponse> exception(CertificationBodyAccessException e) {
         LOGGER.error("Caught ACB access exception.", e);
         return new ResponseEntity<ErrorResponse>(
                 new ErrorResponse(e.getMessage() != null ? e.getMessage() : "Unauthorized ACB Access."),
                 HttpStatus.FORBIDDEN);
     }
 
     @ExceptionHandler(MissingReasonException.class)
-    public ResponseEntity<ErrorResponse> exception(final MissingReasonException e) {
+    public ResponseEntity<ErrorResponse> exception(MissingReasonException e) {
         LOGGER.error("Caught missing reason exception.", e);
         return new ResponseEntity<ErrorResponse>(
                 new ErrorResponse(e.getMessage() != null ? e.getMessage()
                         : "A reason is required to perform this action."),
                 HttpStatus.BAD_REQUEST);
     }
 
+    @ExceptionHandler(HttpClientErrorException.class)
+    public ResponseEntity<ErrorResponse> exceptionHandler(HttpClientErrorException e, HttpServletRequest request) {
+        LOGGER.error("HttpClientErrorException: " + e.getMessage(), e);
+        return new ResponseEntity<ErrorResponse>(
+                new ErrorResponse(e.getMessage() != null ? e.getMessage()
+                        : "The request could not be completed. Please check your credentials and the connection to the server."),
```

Comment:
It's working. The "e.getMessage()" value from Jira is literally "401". I preferred to send back the error message from Jira thinking it would be more useful. I can change up this logic to just always send back a string of our choosing if you would prefer.

Reply:
Meh. Maybe take a look at it with the tickets in the backlog about "what to do when it doesn't respond" stuff. I think it's good now